### PR TITLE
Property creation and manage listing

### PR DESCRIPTION
**Pull request by aplazand20**
[Commands]
	1) docker-compose down 
       2) docker-compose up --build -d 
       3) docker-compose exec web rails db:migrate:reset 

[Pre-conditions]
 • Login as a user
 • Click the “Become a host” button located on the header
 • Fill in the forms

[Expected Output]
 •  User can name the property and describe it
 •  User will pick a home type 
 •  User will pick a room type 
 •  User will pick how many bedrooms 
 •  User will pick how many bathrooms 
 •  User will pick how many guests the home can accommodate 
 •  User can pick amenitites 
 •  User can declare a price 
 •  User can declare the location of the property
 •  Manage property listing – manage as in able to update and see the changes on the database

[Notes]
 • I will improve the UI once I’m done with all the functionalities
 •  Publish button cannot be of use as of now because I have to be able to upload an image first which is the next task

### CODE REVIEW

File: app/assets/stylesheets/custom.scss [hunk 1]
```diff
@@ -28,11 +28,22 @@ body {
     font-weight: 400;
```
Comment (charlotteds) [inline app/assets/stylesheets/custom.scss]: Use 2-spaces for indenting css code.

File: app/controllers/properties_controller.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
+class PropertiesController < ApplicationController
+  def new
+    @property = current_user.properties.build
+  end
+
+  def create
+    @property = current_user.properties.build(property_params)
+     if @property.save 
+      redirect_to listing_property_url(@property)
```
Comment (charlotteds) [inline app/controllers/properties_controller.rb:12]: Add message for successful save.

File: app/controllers/properties_controller.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  def amenities
+    @property = Property.find(params[:id])
+  end
+
+  def location
+    @property = Property.find(params[:id])
+  end
+
+  private
+      def property_params
```
Comment (charlotteds) [inline app/controllers/properties_controller.rb]: Fix indention for method `property_params.`

File: app/controllers/properties_controller.rb [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      @property.update(property_params)
+      redirect_to request.referrer
+    end
+  end
+
+  def show
+  end
+
+  def listing
+    @property = Property.find(params[:id])
```
Comment (charlotteds) [inline app/controllers/properties_controller.rb]: Create a filter method for setting property: `@property = Property.find(params[:id])`

File: app/controllers/properties_controller.rb [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    puts @property.errors.full_messages
+  end
+
+  def edit
+    @property = Property.find(params[:id])
+  end
+
+  def update
+    @property = Property.find(params[:id])
+    if @property.save
```
Comment (charlotteds) [inline app/controllers/properties_controller.rb]: Condition should be if `@property` is updated

File: app/test.html.erb [hunk 6]
```diff
@@ -0,0 +1,65 @@
+<div class="row justify-content-center">
```
Comment (charlotteds) [inline app/test.html.erb]: What is this file for?

File: app/views/properties/_amenities.html.erb [hunk 7]
```diff
@@ -0,0 +1,35 @@
+<%= form_for(@property) do |f| %>
+<div class="row justify-content-center">
```
Comment (charlotteds) [inline app/views/properties/_amenities.html.erb]: Indent `form` contents

File: app/views/properties/_location.html.erb [hunk 8]
```diff
@@ -0,0 +1,19 @@
+<%= form_for(@property) do |f| %>
```
Comment (charlotteds) [inline app/views/properties/_location.html.erb]: Indent `form` contents

File: app/views/properties/_photos.html.erb [hunk 9]
```diff
@@ -0,0 +1,18 @@
+<%= form_for(@property) do |f| %>
```
Comment (charlotteds) [inline app/views/properties/_photos.html.erb]: Indent `form` contents

File: app/views/properties/_pricing.html.erb [hunk 10]
```diff
@@ -0,0 +1,19 @@
+<%= form_for(@property) do |f| %>
+<div class="row justify-content-center">
```
Comment (charlotteds) [inline app/views/properties/_pricing.html.erb]: Indent `form` contents

**Review by charlotteds — COMMENTED** [review]
Notes:

- At http://localhost:3002/properties/new
    - If user tries to click `Create my room` with empty input, **display error message**
- **Display success message** when a property information is saved

- Publish button should be disabled for current property status
- No need to create partial views for the following:
    - Amenities
    - Description
    - Location
    - Photos
    - Pricing
- Unchanged files must not be included in commit

**Comment by aplazand20** [comment]
> No need to create partial views for the following:
> 
>     * Amenities
> 
>     * Description
> 
>     * Location
> 
>     * Photos
> 
>     * Pricing

You mean remove their views?

File: app/controllers/application_controller.rb [hunk 11]
```diff
@@ -5,6 +5,10 @@ class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
   before_action :configure_permitted_parameters, if: :devise_controller?
 
+  def properties_param
```
Comment (charlotteds) [inline app/controllers/application_controller.rb]: Create `before_action` in `PropertiesController`

**Review by charlotteds — COMMENTED** [review]
Notes:
> Create a filter method for setting property: `@property = Property.find(params[:id])`

- With the comment above, I wanted you to use a `before_action` filter within your properties controller to omit the redundancy of calling `@property = Property.find(params[:id])` every time it is needed in a method.

- **Delete** `photos/images` folder then **add** `/public/system/photos/images` to gitignore file before pushing

**Review by charlotteds — APPROVED** [review]

**Pull request merged**